src/app/core/services/storage.service.ts

```diff
@@ -171,19 +171,12 @@ export class StorageService {
   }
 
   getAssessment(task: Task) {
-    const defaultAssessment = this.get(StorageKeys.CONFIG_ASSESSMENTS)
-    const clinicalAssesment = this.get(StorageKeys.CONFIG_CLINICAL_ASSESSMENTS)
-    return Promise.all([defaultAssessment, clinicalAssesment]).then(
-      assessments => {
-        for (let i = 0; i < assessments.length; i++) {
-          for (let j = 0; j < assessments[i].length; j++) {
-            if (assessments[i][j].name === task.name) {
-              return assessments[i][j]
-            }
-          }
-        }
-      }
-    )
+    return Promise.all([
+      this.get(StorageKeys.CONFIG_ASSESSMENTS),
+      this.get(StorageKeys.CONFIG_CLINICAL_ASSESSMENTS)
+    ]).then(([nonClinical, clinical]) => {
+      return nonClinical.concat(clinical).find(a => a.name == task.name)
```

Comment:
What would be the real world value of task name? Can two tasks have the same name?

Reply:
Yup, two tasks should not have the same name since they are in one `protocol.json` file. I will add some documentation on this.